boltdesignsystem/bolt
# DS-140: setup "elements" and "layouts" namespaces

## Jira

https://pegadigitalit.atlassian.net/browse/DS-140

## Summary

Setup initial "elements" and "layouts" namespaces in Bolt. Add _private_ Link element and List layout as starting point.

## Details

For the most part, adding "elements" and "layouts" requires no config changes, just adding new packages with the appropriate name, e.g. `@bolt/elements-*`. The only config change required was Jest, adding `elements` and `layouts` to the list of directories that are compiled by Babel before running tests.

At the moment, "elements" and "layouts" are top-level directories within the "packages" folder. It would be nicer if they were grouped together with "components", but as that would require moving _every single component_ I suggest we wait and reevaluate once we're ready to ship our first "element" and/or "layout".

Note:
- The Link (element) and List (layout) packages are private for now, so this is safe to merge.
- There is not yet a corresponding Pattern Lab section for these new namespaces, will be handled separately.

## How to test

- Checkout branch, run `yarn setup`, and verify site builds and runs properly.
- Try changing one of the Link Twig demos to use the `@bolt-elements-link/link.twig` template and verify it works.
